lora/models.py

```diff
@@ -32,14 +90,58 @@ def from_linear(linear: nn.Linear, rank: int = 8):
         lora_lin.linear = linear
         return lora_lin
 
+    def to_linear(self):
+        linear = self.linear
+        bias = "bias" in linear
+        weight = linear.weight
+        is_quantized = isinstance(linear, nn.QuantizedLinear)
+
+        # Use the same type as the linear weight if not quantized
+        dtype = weight.dtype
+
+        if is_quantized:
+            dtype = mx.float16
```

Comment:
This could be `dtype = linear.scales.dtype` but I think it might be better to just fix it to `float16` for quantized layers anyway.